### PR TITLE
Add campaign_spend_fractions_generator to SystemParameters

### PR DESCRIPTION
**Pull request by jiayu-google**

### CODE REVIEW

File: src/simulator/system_parameters.py [hunk 1]
```diff
@@ -40,21 +41,35 @@ class SystemParameters(NamedTuple):
...
-    liquid_legions: LiquidLegionsParameters
-    generator: Generator
+    campaign_spend_fractions: List[float] = [0]
+    liquid_legions: LiquidLegionsParameters = LiquidLegionsParameters()
+    generator: np.random.Generator = np.random.default_rng(1)
+    campaign_spend_fractions_generator: Callable[[int], List[float]] = lambda x: None
 
     def __str__(self) -> str:
         spend_str = ",".join([f"{s}" for s in self.campaign_spend_fractions])
+        # TODO(jiayu): add str for campaign_spend_fractions_generator if necessary
```
Comment (matthewclegg) [inline src/simulator/system_parameters.py:72]: I don't think this would be necessary, as long as we keep the campaign_spend_fractions in the string.

Comment (matthewclegg) [inline src/simulator/system_parameters.py:72]: I don't think this would be necessary, as long as we keep the campaign_spend_fractions in the string.

**Review by matthewclegg — APPROVED** [review]
(1) Does it make sense to eliminate the campaign_spend_fractions argument?

(2) I wonder if the campaign_spend_fractions_generator should take an numpy.random.Generator argument to allow for the possibility of generating random spend fractions?

File: src/simulator/system_parameters.py (hunk 1 above)
Comment (jiayu-google) [reply src/simulator/system_parameters.py]: makes sense. deleted the todo item.

**Review by RonPinelliMRC — APPROVED** [review]
MRC approving based on cursory review after confirming non-independent code review on an exception basis. Experimental code will require independent review prior to publication of or reliance on results generated.

**Pull request merged**